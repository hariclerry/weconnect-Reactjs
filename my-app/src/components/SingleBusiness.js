import React from 'react';
import './static/css/bootstrap.min.css'

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Loader from 'react-loader';
// import { NavLink } from 'react-router-dom';

// TO DO:import { NotificationManager } from "react-notifications";
import { singleBusiness, deleteBusiness } from "../actions/businessActions";
import {checkIfUserIsLoggedIn} from '../actions/userActions';
import NavBar from "./NavBar";
import Footer from './Footer';


class SingleBusiness extends React.Component {

       state = {
         loaded: false
       }
      componentWillMount() {
        
        checkIfUserIsLoggedIn(this.props.email,this.props.history);
        const { id } = this.props.match.params;
        console.log(id);
        this.props.singleBusiness(id).then(()=>{});
        this.setState({loaded: true})
      }

      deleteBusiness = id  =>{
        if (window.confirm('Are you sure you want to delete this business?')) {
                this.props.deleteBusiness(id).then(()=>{
                  console.log(" confirmed deleted........")
                });
               this.props.history.push("/dashboard");
        }
    }
    render () {
        // Deconstructing an object
        const{business} = this.props;
        const {loaded} = this.state;

        return (
    <div>
    <NavBar />
            <div className="container">
            <h4>Hello User</h4>
            <hr/>

        <div className="col-md-8 col-md-offset-2">
        <Loader loaded={loaded}>
        <h1>{business.name}</h1>
        <p><b>Category:   </b> {business.category}</p>
        <p><b>Location:   </b>{business.location}</p>
        <p><b>Description:  </b>{business.description}</p>

        <a href="/addreview" className="btn btn-info" role="button" style={{margin: "2%"}}>Review</a>
        <a href="/edit" className="btn btn-info" role="button" color="green" style={{margin: "2%"}}>Edit</a>
        <button onClick={() => this.deleteBusiness(business.id) } className="btn btn-default"  role="button" style={{margin: "2%"}}> Delete </button>
        </Loader>
      </div>

     </div>

    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <Footer />
     </div>

        )
    }
}
SingleBusiness.protoTypes = {
    email: PropTypes.string.isRequired,
    singleBusiness: PropTypes.func.isRequired,
    deleteBusiness: PropTypes.func.isRequired
  };

function mapStateToProps(state, ownProps){
  console.log(state.business.singleBusinessMessage.business_data);
   return {
    email: state.auth.loginData.email,
    business: state.business.singleBusinessMessage.business_data,
    businessId: ownProps.match.params.id,
    deletedBusiness: state.business.deleteBusinessMessage,

   }

  };

  // const mapDispatchToProps = dispatch => ({

  //     loadBusiness: (businessId) => dispatch(singleBusiness(businessId))
  // });

  export default withRouter(connect(mapStateToProps, {singleBusiness, deleteBusiness})(SingleBusiness));