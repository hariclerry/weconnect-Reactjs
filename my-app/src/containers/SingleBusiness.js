import React from "react";

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
// import { NavLink } from 'react-router-dom';

// TO DO:import { NotificationManager } from "react-notifications";
import SingleBusiness from "../components/SingleBusiness";
import { singleBusiness, deleteBusiness } from "../actions/businessActions";
import { checkIfUserIsLoggedIn } from "../actions/userActions";

class Business extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loaded: false
    };
  }

  componentDidMount() {
    checkIfUserIsLoggedIn(this.props.email, this.props.history);
    const { id } = this.props.match.params;
    this.props.singleBusiness(id);
    // this.setState({loaded: true})
  }

  deleteBusiness = id => {
    if (window.confirm("Are you sure you want to delete this business?")) {
      this.props.deleteBusiness(id).then(() => {
        console.log(" confirmed deleted........");
      });
      this.props.history.push("/dashboard");
    }
  };
  render() {
    const { business } = this.props;
    const { loaded } = this.state;
    return (
      <SingleBusiness
        business={business}
        deleteBusiness={this.deleteBusiness}
        loaded={loaded}
      />
    );
  }
}
SingleBusiness.protoTypes = {
  email: PropTypes.string.isRequired,
  singleBusiness: PropTypes.func.isRequired,
  deleteBusiness: PropTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => {
  console.log(state);
  return {
    email: state.auth.loginData.email,
    business: state.business.singleBusinessMessage.business_data,
    businessId: ownProps.match.params.id,
    deletedBusiness: state.business.deleteBusinessMessage
  };
};

const mapDispatchToProps = dispatch => ({
  singleBusiness: id => dispatch(singleBusiness(id)),
  deleteBusiness: id => dispatch(deleteBusiness(id))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Business)
);
