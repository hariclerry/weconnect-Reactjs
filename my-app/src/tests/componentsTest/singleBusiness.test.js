import React from "react";
import {shallow, mount} from "enzyme";
import { MemoryRouter } from "react-router-dom";
import jwt from "jsonwebtoken";
import weconnectStore from "store";
import SingleBusiness from "containers/business/singleBusiness";

const loginUserMock = { access_token: jwt.sign({ email: "clerry@gmail.com", password: "this199" }, "hard to guess string")}

describe('SingleBusiness component',() =>{
  beforeEach(() => {
    localStorage.setItem("token", loginUserMock.access_token);
});
it('should match snapshot',() =>{
  const wrapper = shallow(<SingleBusiness />)
  expect(wrapper).toMatchSnapshot();
})

it("should render the SingleBusiness component", () => {
  const mountComponent = shallow(<MemoryRouter>< SingleBusiness store = {weconnectStore}/></MemoryRouter>)
  expect(mountComponent.length).toBe(1);
  
});

}

)