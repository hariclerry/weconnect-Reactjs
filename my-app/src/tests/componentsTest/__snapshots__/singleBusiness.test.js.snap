// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business component should match snapshot 1`] = `<Route />`;

exports[`RegisterBusiness component should match snapshot 1`] = `<Route />`;

exports[`SBusiness component should match snapshot 1`] = `<Route />`;

exports[`SingBusiness component should match snapshot 1`] = `<Route />`;

exports[`SingleBusiness component should match snapshot 1`] = `<Route />`;
